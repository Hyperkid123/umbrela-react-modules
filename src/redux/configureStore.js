import {createStore, applyMiddleware, compose} from 'redux';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import { initialize, setActiveLanguage, addTranslationForLanguage } from 'react-localize-redux';
import en from '../lang/en.js';
import cs from '../lang/cs.js';

const loggerMiddleware = createLogger();
const composeEnhacer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default function configureStore(preloadedState) {
  const languages = [{
    name: 'Czech',
    code: 'cs',
  },{
    name: 'English',
    code: 'en',
  }];
  let store;
  if(process.env.NODE_ENV !== 'production') {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
          loggerMiddleware
        )
      )
    )
  } else {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
        ),
      )
    )
  }
  store.dispatch(initialize(languages))
  store.dispatch(addTranslationForLanguage(en, 'en'));
  store.dispatch(addTranslationForLanguage(cs, 'cs'));
  store.dispatch(setActiveLanguage(window.lang));
  return store;
}
