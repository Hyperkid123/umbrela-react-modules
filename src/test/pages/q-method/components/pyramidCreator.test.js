import React from 'react'
import PyramidCreator, {PyramidCreator as Snapshot} from '../../../../pages/q-method/components/pyramidCreator';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'

describe('Q-method navigation component', () => {
  const initialState  = {
    qMethodBuilder: {
      present: {
        rows: [{
          blocks: 1
        },{
          blocks: 3
        }],
      },
      past: [],
      future: [],
    }
  };

  const mockStore = configureStore();
  it('should create component correctly', () => {
    let store = mockStore(initialState);
    let wrapper = shallow(<PyramidCreator store={store}/>);
    expect(wrapper).toBeDefined(); 
  })
  it('should render correclty' , () => {
      const tree = shallow(
        <Snapshot rows={[{blocks: 2}, {blocks: 5}]}/>
      ).dive();
      expect(toJson(tree)).toMatchSnapshot();
  });
});
