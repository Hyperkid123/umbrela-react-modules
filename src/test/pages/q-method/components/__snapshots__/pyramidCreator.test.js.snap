// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Q-method navigation component should render correclty 1`] = `
<div
  className="sc-bdVaJa bOSErT"
>
  <PyramidDescription />
  <styled.div>
    <styled.div
      column={true}
    >
      <PyramidRow
        blocks={2}
        key="0"
        rowIndex={0}
      />
      <PyramidRow
        blocks={5}
        key="1"
        rowIndex={1}
      />
    </styled.div>
  </styled.div>
  <RowHandler
    rows={2}
  />
</div>
`;
